### PR TITLE
Wip nitzan fixing few rados/test.sh

### PR DESCRIPTION
**Pull request by NitzanMordhai**
We have few issues that keep pop while running rados/test.sh in https://tracker.ceph.com/issues/50042 
Most of them related to watch timeout that causing tests to fail, some of them related to Segmentation fault during handle_error.

Some of the test will fail if https://github.com/ceph/ceph/pull/48641 is not present.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by NitzanMordhai** [comment]
jenkins test make check

**Comment by NitzanMordhai** [comment]
jenkins test windows

**Comment by NitzanMordhai** [comment]
jenkins test make check arm64

**Comment by NitzanMordhai** [comment]
jenkins test windows

**Comment by NitzanMordhai** [comment]
jenkins test make check arm64

**Comment by rzarzynski** [comment]
@badone: Hello Brad! Would you mind taking a look?

**Review by badone — APPROVED** [review]
This looks good to me. Thanks Nitzan.

File: src/test/librados/watch_notify.cc [hunk 1]
```diff
@@ -90,7 +90,9 @@ void LibRadosWatchNotify::watch_notify2_test_errcb_reconnect(void *arg,
   auto thiz = reinterpret_cast<LibRadosWatchNotify*>(arg);
   ceph_assert(thiz);
   thiz->notify_err = rados_unwatch2(thiz->ioctx, cookie);
-  thiz->notify_cookies.erase(cookie); //delete old cookie
+  if (thiz->notify_cookies.count(cookie)) {
+    thiz->notify_cookies.erase(cookie);
+  }
```
Comment (tchaikov) [inline src/test/librados/watch_notify.cc:95]: > tests: Segmentation fault during handle_error

the title reads like a bug report. it is supposed to be a summary of the change made in the commit.

and i think

```c++
thiz->notify_cookies.erase(cookie);
```

and 

```c++
if (thiz->notify_cookies.count(cookie)) {
     thiz->notify_cookies.erase(cookie);
   }
```

are equivalent. i am not sure what you mean by "notify_cookies set is not initialized yet". if it's just empty, then `thiz->notify_cookies.erase(cookie)` is a noop.

**Review by tchaikov — CHANGES_REQUESTED** [review]

File: src/test/system/systest_runnable.cc [hunk 2]
```diff
@@ -97,8 +97,10 @@ start()
     #else
     std::string err_msg;
     ret = preforker.prefork(err_msg);
-    if (ret < 0)
-      preforker.exit(ret);
+    if (ret < 0) {
```
Comment (tchaikov) [inline src/test/system/systest_runnable.cc:100]: might want to fix the typo in the title of the commit message.

Comment (tchaikov) [reply src/test/system/systest_runnable.cc:100]: the title of the commit message reads like a bug report. ideally, the title of the commit message should use the imperative mood. see also https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes . please use `git commit --amend`  and `git push -f` for revising the commit message and its title.

File: src/test/librados/TestCase.cc [hunk 3]
```diff
@@ -13,7 +14,8 @@ rados_t RadosTestNS::s_cluster = NULL;
 
 void RadosTestNS::SetUpTestCase()
 {
-  pool_name = get_temp_pool_name();
+  auto PoolPrefix = fmt::format("{}_", ::testing::UnitTest::GetInstance()->current_test_case()->name());
```
Comment (tchaikov) [inline src/test/librados/TestCase.cc:17]: please following our naming convention. `PoolPrefix` should be `pool_prefix`.

File: src/test/system/systest_runnable.cc (hunk 2 above)
Comment (tchaikov) [reply src/test/system/systest_runnable.cc:100]: the same applies to other commits in this PR.

File: src/test/librados/watch_notify.cc (hunk 1 above)
Comment (NitzanMordhai) [reply src/test/librados/watch_notify.cc]: Right, just checked it, i thought that's the reason for the seg. Fault 
I'm going to remove that commit, will need to address that if we will hit it again on different test.

File: src/test/system/systest_runnable.cc (hunk 2 above)
Comment (NitzanMordhai) [reply src/test/system/systest_runnable.cc:100]: Done

File: src/test/librados/TestCase.cc (hunk 3 above)
Comment (NitzanMordhai) [reply src/test/librados/TestCase.cc]: Done

File: src/test/librados/test_shared.h [hunk 4]
```diff
@@ -21,7 +21,7 @@ class TestAlarm
 public:
   #ifndef _WIN32
   TestAlarm() {
-    alarm(1200);
+    alarm(2401);
```
Comment (tchaikov) [inline src/test/librados/test_shared.h:24]: the title reads 

> cxx tests should be more readable

how does this change improve the readability?

again, the title of the commit message reads like a bug report. ideally, the title of the commit message should use the imperative mood. see also https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes . please use `git commit --amend`  and `git push -f` for revising the commit message and its title.

File: src/test/librados/watch_notify.cc [hunk 5]
```diff
@@ -137,11 +137,14 @@ TEST_F(LibRadosWatchNotify, WatchNotify) {
   char buf[128];
   memset(buf, 0xcc, sizeof(buf));
   ASSERT_EQ(0, rados_write(ioctx, "foo", buf, sizeof(buf), 0));
+  std::cout << "object write complete" << std::endl;
```
Comment (tchaikov) [inline src/test/librados/watch_notify.cc:140]: i'd drop this change. for a test, no news is good news.

File: src/test/librados/watch_notify.cc [hunk 6]
```diff
@@ -137,11 +137,14 @@ TEST_F(LibRadosWatchNotify, WatchNotify) {
...
   uint64_t handle;
   ASSERT_EQ(0,
       rados_watch(ioctx, "foo", 0, &handle, watch_notify_test_cb, NULL));
+  std::cout << "object watch added " << handle << std::endl;
```
Comment (tchaikov) [inline src/test/librados/watch_notify.cc:144]: ditto.

File: src/test/librados/watch_notify.cc [hunk 7]
```diff
@@ -137,11 +137,14 @@ TEST_F(LibRadosWatchNotify, WatchNotify) {
...
   for (unsigned i=0; i<10; ++i) {
     int r = rados_notify(ioctx, "foo", 0, NULL, 0);
+    std::cout << "rados notify return:" << r << std::endl;
```
Comment (tchaikov) [inline src/test/librados/watch_notify.cc:147]: ditto.

File: src/test/librados/test_shared.h (hunk 4 above)
Comment (NitzanMordhai) [reply src/test/librados/test_shared.h:24]: Moving that change to new commit

File: src/test/librados/watch_notify.cc (hunk 5 above)
Comment (NitzanMordhai) [reply src/test/librados/watch_notify.cc]: done

File: src/test/librados/watch_notify.cc (hunk 6 above)
Comment (NitzanMordhai) [reply src/test/librados/watch_notify.cc]: done

File: src/test/librados/watch_notify.cc (hunk 7 above)
Comment (NitzanMordhai) [reply src/test/librados/watch_notify.cc]: done

**Review by tchaikov — APPROVED** [review]

**Comment by ljflores** [comment]
Rados suite review: https://pulpito.ceph.com/?branch=wip-yuri-testing-2023-01-23-0926

Failures, unrelated:
    1. https://tracker.ceph.com/issues/58587 -- new tracker
    2. https://tracker.ceph.com/issues/58585
    3. https://tracker.ceph.com/issues/58098 -- fix merged to latest main
    4. https://tracker.ceph.com/issues/58256 -- fix merged to latest main
    5. https://tracker.ceph.com/issues/57900
    6. https://tracker.ceph.com/issues/58475
    7. https://tracker.ceph.com/issues/58560

Details:
    1. test_dedup_tool.sh: test_dedup_object fails when pool 'dedup_chunk_pool' does not exist - Ceph - RADOS
    2. rook: failed to pull kubelet image - Ceph - Orchestrator
    3. qa/workunits/rados/test_crash.sh: crashes are never posted - Ceph - RADOS
    4. ObjectStore/StoreTestSpecificAUSize.SpilloverTest/2: Expected: (logger->get(l_bluefs_slow_used_bytes)) >= (16 * 1024 * 1024), actual: 0 vs 16777216 - Ceph - RADOS
    5. mon/crush_ops.sh: mons out of quorum - Ceph - RADOS
    6. test_dashboard_e2e.sh: Conflicting peer dependency: postcss@8.4.21 - Ceph - Mgr - Dashboard
    7. test_envlibrados_for_rocksdb.sh failed to subscrib repo - Ceph

**Comment by kamoltat** [comment]
Encountered this error in: yuriw-2023-03-02_00:09:05-rados-wip-yuri11-testing-2023-03-01-1424-distro-default-smithi/7191503 PTAL

The tracker: https://tracker.ceph.com/issues/49524 points to this PR @NitzanMordhai

**Pull request merged**